test_conformance/extensions/cl_khr_semaphore/test_semaphores.cpp

```diff
@@ -586,25 +587,59 @@ int test_semaphores_multi_wait(cl_device_id deviceID, cl_context context,
 int test_semaphores_queries(cl_device_id deviceID, cl_context context,
                             cl_command_queue defaultQueue, int num_elements)
 {
-    cl_int err;
+    cl_int err = CL_SUCCESS;
 
-    if (!is_extension_available(deviceID, "cl_khr_semaphore"))
+    cl_platform_id platform_id = 0;
+    cl_uint num_devices = 0;
+    // find out what platform the harness is using.
+    err = clGetDeviceInfo(deviceID, CL_DEVICE_PLATFORM, sizeof(cl_platform_id),
+                          &platform_id, nullptr);
+    test_error(err, "clGetDeviceInfo failed");
+
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_ALL, 0, nullptr,
+                         &num_devices);
+    test_error(err, "clGetDeviceIDs failed");
+
+    std::vector<cl_device_id> devices(num_devices);
+    cl_device_id capable_device = deviceID;
+
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_ALL, num_devices,
+                         devices.data(), nullptr);
+    test_error(err, "clGetDeviceIDs failed");
+
+    // if possible use device other than the one from harness
+    for (auto device : devices)
+    {
+        if (deviceID != device
+            && is_extension_available(capable_device, "cl_khr_semaphore"))
+        {
+            capable_device = device;
+            break;
+        }
+    }
+
+    if (deviceID == capable_device
+        && !is_extension_available(capable_device, "cl_khr_semaphore"))
     {
         log_info("cl_khr_semaphore is not supported on this platoform. "
```

Comment:
Consider fixing this typo (platoform -> platform) since you're editing this file.
However, given that I see no other issues with this patch, you may want to just push and fix the typo at the next opportunity.

Reply:
Of course, done!